### PR TITLE
chore(test-data): create additional attribute type models, other fixes

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

When creating product types, we discovered that attribute types were not being transformed to match the expected graphql draft shape. This PR creates three new [product type attribute models](https://docs.commercetools.com/api/projects/productTypes#attributeenumtype) (Enum, Text, and Boolean) with their own draft models that will transform to graphql correctly.

It also makes a minor change to the customer draft presets, noted inline.

### CODE REVIEW

File: models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  ),
+  graphql: Transformer<
+    TAttributeBooleanTypeDraft,
+    TAttributeBooleanTypeDraftGraphql
+  >('graphql', {
+    buildFields: [],
+    replaceFields: ({ fields }) => ({
+      boolean: {
+        dummy: fields.name,
```
Comment (jaikamat) [inline models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts:21]: Graphql expects this field oddly enough. I assumed passing along any string would be fine.

File: models/product-type/src/product-type-draft/presets/sample-data-fashion/jackets.spec.ts [hunk 2]
```diff
@@ -106,9 +106,7 @@ describe(`with jackets preset`, () => {
             ],
             "name": "sleeve_length",
             "type": {
-              "__typename": "enum",
-              "name": "enum",
-              "values": [
+              "enum": [
```
Comment (jaikamat) [inline models/product-type/src/product-type-draft/presets/sample-data-fashion/jackets.spec.ts:109]: It's easier to see what graphql expects when comparing the diff in the snapshot tests 👍

File: models/customer/src/constants.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const authenticationMode = {
+  ExternalAuth: 'ExternalAuth',
```
Comment (tylermorrisford) [inline models/customer/src/constants.ts:2]: <3

**Review by tylermorrisford — APPROVED** [review]

File: models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts (hunk 1 above)
Comment (stephsprinkle) [reply models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts:21]: I see this but what is it for??? (probably just shouting frustration into the void)

File: models/attribute-type/src/attribute-enum-type/attribute-enum-type-draft/generator.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Generator } from '@commercetools-test-data/core';
+import { TAttributeEnumTypeDraft } from '../types';
+
+// https://docs.commercetools.com/api/projects/productTypes#attributeenumtype
+
+export const generator = Generator<TAttributeEnumTypeDraft>({
+  fields: {
+    name: 'enum',
+    values: null,
```
Comment (stephsprinkle) [inline models/attribute-type/src/attribute-enum-type/attribute-enum-type-draft/generator.ts:9]: This is a required field, will null work here?

File: models/attribute-type/src/attribute-enum-type/generator.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Generator } from '@commercetools-test-data/core';
+import { TAttributeEnumType } from './types';
+
+// https://docs.commercetools.com/api/projects/productTypes#attributeenumtype
+
+export const generator = Generator<TAttributeEnumType>({
+  fields: {
+    name: 'enum',
+    values: null,
```
Comment (stephsprinkle) [inline models/attribute-type/src/attribute-enum-type/generator.ts:9]: Same same

File: models/product-type/src/product-type-draft/presets/sample-data-fashion/accessories.ts [hunk 6]
```diff
@@ -18,7 +21,7 @@ const accessories = (): TProductTypeDraftBuilder =>
         .inputTip(LocalizedString.presets.empty()['en-US']('Type'))
         .isRequired(true)
         .type(
-          AttributeType.presets.enumType().values([
```
Comment (stephsprinkle) [inline models/product-type/src/product-type-draft/presets/sample-data-fashion/accessories.ts:21]: Can these attribute type presets be removed?

File: models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts (hunk 1 above)
Comment (jaikamat) [reply models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts:21]: Ah, I see that this field is nullable. I'm going to pass that instead

Comment (jaikamat) [reply models/attribute-type/src/attribute-boolean-type/attribute-boolean-type-draft/transformers.ts:21]: All of which is to say, I have no idea what this field is for either 😆

File: models/attribute-type/src/attribute-enum-type/attribute-enum-type-draft/generator.ts (hunk 4 above)
Comment (jaikamat) [reply models/attribute-type/src/attribute-enum-type/attribute-enum-type-draft/generator.ts:9]: I'll use an empty array instead 👍

File: models/product-type/src/product-type-draft/presets/sample-data-fashion/accessories.ts (hunk 6 above)
Comment (jaikamat) [reply models/product-type/src/product-type-draft/presets/sample-data-fashion/accessories.ts:21]: Good call. Much cleaner. See b822e08a81c1dd8d55fbab268185b23a210696d0

**Pull request merged**